src/common/telemetry/lib/Telemetry.cpp

```diff
@@ -0,0 +1,229 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "Telemetry.hpp"
+#include "ParameterSets.hpp"
+
+#include <algorithm>
+#include <chrono>
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+#include <fstream>
+#include <iostream>
+#include <nlohmann/json.hpp>
+#include <sstream>
+#include <thread>
+
+using namespace MAT;
+
+namespace Telemetry
+{
+
+TelemetryManager::TelemetryManager(bool enableDebug, int teardownTime, OsConfigLogHandle logHandle)
+    : m_log(logHandle)
+    , m_logManager(nullptr)
+    , m_logger(nullptr)
+{
+#if defined(DEBUG) || defined(_DEBUG) || !defined(NDEBUG)
```

Comment:
how is this used? Why is it needed? Could you point to the tests supporting both positive and negative setting of DEBUG and the other variables?

Reply:
It is used, and there are also tests supporting the file argument (TelemetryBinTest::VerboseFlagWorks)